config/site_name: Use hostname from config in emails

justwheel opened the pull request:
This commit changes the site_name dict around in the config file. It
adds two fields: `prefix` and `fqdn`. These are in turn used in the
email template files to always use the configured URLs in places like
email.

There is probably a more elegant way to do this. But I am satisfied
enough with this because it puts the hostname in control of the user,
and so long as we are consistent about pulling the hostname in this way
across the back-end, it poses low risk.

Technically, this is lagging work on #157.

justwheel committed:
config/site_name: Use hostname from config in emails

This commit changes the site_name dict around in the config file. It
adds two fields: `prefix` and `fqdn`. These are in turn used in the
email template files to always use the configured URLs in places like
email.

There is probably a more elegant way to do this. But I am satisfied
enough with this because it puts the hostname in control of the user,
and so long as we are consistent about pulling the hostname in this way
across the back-end, it poses low risk.

Technically, this is lagging work on #157.

Signed-off-by: Justin W. Flory <git@jwf.io>

nlMeminger commented on config.yml.example:
```diff
@@ -60,8 +60,10 @@ outgoing_smtp:
 
-site_name: "localhost"
+site_name:
+  prefix: "http://"
```
hardcoding http like this makes me worried.  Specifically mandating the "://" This should be changed from prefix to protocol with two possible options ['http', 'https'] with some help text above.  then the "://" should be added manually in the backend.

justwheel replied:
@nlMeminger Fair feedback. You challenged me to think of something smarter. I think I addressed this in commit e219d5b001f40c67489551690e581815b1ba7f28. Please take a look and re-review.

nlMeminger commented on grasa_event_locator/templates/messaging/provider_account_approval_mail.txt:
```diff
@@ -3,4 +3,4 @@ Hello,
 Please login below to start adding events:
 
-http://grasa.larrimore.de/login.html
+{{ config.site_name.prefix }}{{ config.site_name.fqdn }}/login.html
```
here is where we should be manually adding the "://".  could be something like 
{{ config.site_name.protocol }}://{{ config.site_name.fqdn }}/login.html

justwheel replied:
Resolved. See commit e219d5b001f40c67489551690e581815b1ba7f28.

justwheel committed:
:lock: config: Rework HTTP/HTTPS as a boolean

This addresses @nlMeminger's feedback in #213 and I think is actually a
more elegant way of doing this. Basically, there is a hidden string
inside of the emails now that checks if HTTPS is set to true. If it is,
it includes an 's' character. Otherwise, Jinja silently ignores it when
writing out the HTML.

Signed-off-by: Justin W. Flory <git@jwf.io>

nlMeminger reviewed (approved):
I think this is a bit more intuitive to use.
